src/storage/txn/scheduler.rs

```diff
@@ -862,7 +870,29 @@ fn process_write_impl(
             );
             let rows = keys.len();
             for k in keys {
-                txn.gc(k, safe_point)?;
+                let gc_info = txn.gc(k, safe_point)?;
+
+                if gc_info.found_versions >= GC_LOG_FOUND_VERSION_THRESHOLD {
+                    info!(
+                        "[region {}] GC found at least {} versions for key {}",
+                        ctx.get_region_id(),
+                        gc_info.found_versions,
+                        k
+                    );
+                }
+                // We may delete only part of the versions a time, which may not beyond the logging
+                // threshold `GC_LOG_DELETED_VERSION_THRESHOLD`.
+                if gc_info.deleted_versions as usize >= GC_LOG_DELETED_VERSION_THRESHOLD
+                    || txn.write_size() >= MAX_TXN_WRITE_SIZE
```

Comment:
I don't think the condition `txn.write_size() >= MAX_TXN_WRITE_SIZE` is ok here.

Reply:
That's ok for current threshold. But in future, if the threshold is large and we cannot reach the threshold in a single batch (txn.write_size() >= MAX_TXN_WRITE_SIZE) I think we still need to log an item here. I will change it to a comment for reminders.